src/compiler/core.ts

```diff
@@ -1904,8 +2003,9 @@ namespace ts {
         const aComponents = getNormalizedPathComponents(a, currentDirectory);
         const bComponents = getNormalizedPathComponents(b, currentDirectory);
         const sharedLength = Math.min(aComponents.length, bComponents.length);
+        const comparer = ignoreCase ? compareStringsCaseInsensitive : compareStringsCaseSensitive;

```

Comment:
Would be easier to read if you just called `compareStrings` inside the loop -- then we wouldn't need a `compareStringsCaseSensitive` function and could just use `===` directly.

Reply:
That's what this did before. However, that means we are evaluating the `ignoreCase` condition for each iteration of the loop. This saves a few cycles.

Reply:
I would think that an indirect function call isn't any faster than an if?

Reply:
I've actually removed `compareStrings` now. Comparisons need to be explicit and intentional:

- `compareValues` now only compares numbers
- `compareStringsCaseSensitive` compares strings using an ordinal case-sensitive comparison
- `compareStringsCaseSensitiveUI` compares strings using a dictionary-ordered case-sensitive comparison using the current UI locale
- `compareStringsCaseInsensitive` compares strings using an ordinal case-insensitive comparison
- `compareStringsCaseInsensitiveUI` compares strings using a dictionary-ordered case-insensitive comparison using the current UI locale